pyramid_swagger/api.py

```diff
@@ -101,26 +101,206 @@ def view_for_api_declaration(request):
+            scheme=target_scheme,
+            hostname=target_url.hostname,
+            path=remote_path,
+            fragment=target_url.fragment,
+        ))
+    else:
+        return urlparse(relative_path)
+
+
+def _relpath(path, start):
+    """Return a relative version of a path
+    NOTE: Code duplicated from Python 2.7.9 implementation because the
+    default implementation available on Python 2.6.9 is bugged.
+    """
+
+    start_list = [x for x in os.path.abspath(start).split(os.path.sep) if x]
+    path_list = [x for x in os.path.abspath(path).split(os.path.sep) if x]
+
+    # Work out how much of the filepath is shared by start and path.
+    i = len(os.path.commonprefix([start_list, path_list]))
+
+    rel_list = [os.path.pardir] * (len(start_list) - i) + path_list[i:]
+
+    return os.path.join(*rel_list)
+
+
+def _get_target(spec, target, current_path=''):
+    # Note: we assume that swagger.json file is always a server-local file
+    if target == '':
+        raise ValueError('Empty target')
+
+    target_url = _get_absolute_link(spec, target, current_path)
+    target_scheme = 'file' if target_url.scheme == '' else target_url.scheme
+
+    if target_scheme == 'file':
+        spec_dir = os.path.dirname(urlparse(spec.origin_url).path)
+        target_url = urlparse('{path}#{fragment}'.format(
+            path=_relpath(target_url.path, spec_dir),
+            fragment=target_url.fragment,
+        ))
+
+    if target_scheme in ['file', 'http', 'https']:
+        return target_url
+    else:
+        raise ValueError(
+            'Invalid target: {target}'.format(target=target)
+        )
+
+
+def _marshal_target(target_url):
+    target_scheme = target_url.scheme
+    if len(target_url.path) > 0 and \
+            target_scheme in ['', 'file', 'http', 'https']:
+        value = _low_level_translate(
+            '{scheme}://{host}{path}#{fragment}'.format(
+                scheme=target_url.scheme if len(target_scheme) > 0 else 'file',
+                host=target_url.hostname if target_url.hostname else '',
+                path=target_url.path,
+                fragment=target_url.fragment,
+            ))
+        return value
+    else:
+        raise ValueError(
+            'Invalid target: {target}'.format(target=urlunparse(target_url))
+        )
+
+
+def _unmarshal_target(marshaled_target):
+    if any(marshaled_target.startswith(x) for x in
+           ('file.', 'http.', 'https.')):
+        result = _low_level_translate(marshaled_target, is_marshaling=False)
+        # Remove the file:// scheme to allow the use of relative paths
+        if not result.startswith('file:///'):
+            result = result.replace('file://', '')
+        return result
+    else:
+        raise ValueError(
+            'Invalid marshaled object: {target}'.format(
+                target=marshaled_target
+            )
+        )
+
+
+def is_a_definition(json_path):
```

Comment:
The general idea was: we have troubles if we are dereferencing recursive structure. Since the only possible recursive structure are the data-model we can prevent the complete dereferencing for those objects.
Every reference to a data model should be dereferenced in the [swagger definitions object](http://swagger.io/specification/#definitionsObject). Reading the [Swagger specification](http://swagger.io/specification/) was possible to identify the objects which referees to a *definition* (as described in the comments).

Have you some suggestion about the level of detail that I should insert on the docstring? Could I put part of this comment into the method documentation.